asserts/device_asserts.go

```diff
@@ -218,6 +222,29 @@ var (
 	classicModelOptional = []string{"architecture", "gadget"}
 )
 
+var almostValidName = regexp.MustCompile("^[a-z0-9-]*[a-z][a-z0-9-]*$")
+
+// validateSnapName checks whether the name can be used as a snap name
+//
+// This function should be synchronized with the reference implementation
+// snap.ValidateName() in snap/validate.go
+func validateSnapName(name string, headerName string) error {
```

Comment:
I found the `almostValidateName` a little peculiar and as I dug in, I indeed found it there (with an explanation). I'm not super fond of duplicating it all here. Is there a reason it's not simply calling existing methods (and exporting them if neccessary)?

Reply:
Yes, we don't want to import the `snap` package in asserts, as it would drag in too much dependencies. This was raised by @pedronis in his first pass. As an easier to land solution, I've copied over the code and added tests.

A follow-up I'm planning to do once this lands is to split various name validation code from `snap` into `snap/name` package. Then we could import it without issues. The thing I was working on is right here: https://github.com/snapcore/snapd/compare/master...bboozzoo:bboozzoo/validate-name-separate-package

Reply:
Ah, allright, so this has been raised and will be addressed, it's fine then. Thanks!